### ISSUE TITLE
When a Community is created via importer, first associated protocol becomes default?

### ISSUE THREAD

**Issue by KelleyCoda** [MukurtuCMS/mukurtucms#77]
Hi, huge fan of the zip importer... just trying to figure things out. 
I have never successfully compiled a zip import, so I'm hoping to clear some things. Is there a resource on how to import this way? Some questions I have...
-- Is "group register" on Community and Protocol forms == to the group members, where if I put the user id of the members (comma or semicolon delimited?), then the users will be added?
-- How do I use the group roles column, especially if I have n+ users in the group register?
-- There is no group roles column on the cultural protocol import template file. What role, if any, will users become if added through the "group register"
-- Does the column "Auto Add Users" relate to auto-adding users from the community, or auto-adding users from the site?
-- 2.05 release notes say, "The zip importer will now handle the case where the files have been placed within a single sub-folder (as many zip archive programs do by default)" Do these files still have to be placed in media type subfolders, such as media/image, media/file, etc? Or can I place every type directly in the media folder?
-- The new zip doesn't have a users import or categories. I know that with Mukurtu 2.0, you don't have to pre-import the categories, but what if I wanted to import users and categories too, would it accept a users.csv or a categories.csv?

Thanks!

**Comment by steve-taylor-wsu** [comment]
Hi Kelley,

* "Group register" is the "Group register" boolean field in the advanced tab of the community edit form ("Do not show on registration page" or "Show on registration page"). It won't do anything with users. Actually at the moment, the zip importer doesn't do much with users. Its goal is to get DH items in and out.

* Currently protocols with "Automatically add new users to this protocol" enabled will add any users created in the future to that protocol, regardless of community (does not impact existing users).

* The zip importer doesn't need the media directory, it just needs valid paths. The exporter uses the 'media/type/id' folder structure to help prevent file naming collisions. As long as you use the replacement token '[mukurtu:batch_import_temp_directory]' in the CSV file any path should work (e.g., "[mukurtu:batch_import_temp_directory]/myimagefile.jpg" would be a file named "myimagefile.jpg" in the root of the ZIP archive).

* The zip importer currently does not currently support importing users or categories. The original scope of the zip importer was to only support importing zip files from the Mukurtu exporter. It's clear now that many people want to use it manually for other purposes, so I'm sure we'll be circling back around to expand what it does and to make it more user friendly. This is also why there is so little existing documentation at the moment for it.

Thanks!

**Comment by KelleyCoda** [comment]
Awesome response, thanks! Unfortunately, the importer makes root admin the creator of all of the groups, so it would be nice to be able to manipulate that at least.

**Comment by KelleyCoda** [comment]
Sorry, final question and what prompted this issue: I noticed when I imported the test zip file, the protocol in the protocol sheet (which was an open protocol), became the default protocol for the community. So when I added a user to the community, that user was automatically added to the protocol. Do you know what dictates the automatic membership? Is the script looking for the first child protocol and just adding new community members to that? If so, then I would put the protocol I want to become the default above any other protocols in the csv. Is that correct?

**Comment by steve-taylor-wsu** [comment]
Short answer: Yes, if you are importing a protocol for a community that has no existing protocols, that protocol will become the default protocol. Feeds will import in the order given in the spreadsheet, so put the desired default first.

Longer answer: When creating a new community through the web interface, the first, default protocol is automatically created (creation of My Test Community creates My Test Community Community Only protocol). When importing, this automatic creation does not occur because presumably you are going to import protocols as well, one of which is probably a pre-existing default protocol. Internally there is no flag or indicator for the default protocol, only an assumption that it is at the first position in the data structure. I think this will need to be changed in the future to make it more clear which is the default and to not rely on ordering. Also adding the ability to change which is the default would probably be nice. Let me know if you've got any specific thoughts/ideas on this.

Thanks!

**Comment by KelleyCoda** [comment]
Yeah, it would be a nice feature to be able to select one or more "default" protocols for community members to access. Not sure if the value added is significant. You might consider creating some prototypes and having users test. I can see the advantage to including an open access protocol to every community (and in going back through development documentation, that is what I initially envisioned), but there could be drawbacks and arguments against it.
